src/main/java/ee/tuleva/onboarding/auth/principal/Person.java

```diff
@@ -7,4 +7,6 @@
   String getFirstName();
 
   String getLastName();
+
+  String getPhoneNumber();
```

Comment:
I would not add the `phoneNumber` field to the `Person` since it's totally valid that a person might not have a phone number. You can see the problem with this change also in the places where you're forced to add this Override:
```
  @Override
  public String getPhoneNumber() {
    return null;
  }
  ```
This is a clear indication that we have a problem with our OOP design.

I would maybe introduce a new interface `PersonWithPhone extends Person` or create a separate interface `WithPhoneNumber`, so you could do stuff like `AuthenticatedPerson implements Person, WithPhoneNumber`.

Any other thoughts on how to model this properly in OOP?

Reply:
Maybe using the `Strategy` pattern could work in this case? https://medium.com/@shantanutripathi/strategy-pattern-the-first-design-pattern-to-understand-634e9adab243